stuffer/s2n_stuffer.c

```diff
@@ -55,6 +55,15 @@ int s2n_stuffer_init(struct s2n_stuffer *stuffer, struct s2n_blob *in)
     stuffer->tainted = 0;
     return S2N_SUCCESS;
 }
+
+int s2n_stuffer_init_with_data(struct s2n_stuffer *stuffer, struct s2n_blob *in)
+{
+    GUARD(s2n_stuffer_init(stuffer, in));
+    stuffer->write_cursor = stuffer->high_water_mark = in->size;
```

Comment:
Are you sure you want another place where we'd have to update if there's any changes in the core stuffer logic? Could just cal s2n_stuffer_write() here, or skip_write()

Reply:
If s2n_stuffer_write() is used then an overlapping memcpy() occurs which is undefined behavior that issue #1603 is trying to address. skip_write() could be used yes but then there is an unnecessary call made to s2n_stuffer_reserve_space(). I don't think the possible future update cost would be too great.

Reply:
@rday

Reply:
The current pattern of calling `s2n_stuffer_write` also incurs this extra call to `s2n_stuffer_skip_write`. We shouldn't see any performance degradation, but we would have a lower maintenance burden later on. We still get the UB win by having the `s2n_stuffer_init_with_data` function.

Thanks! This will be a nice addition to stuffer libs!